ELVM.md

```diff
@@ -0,0 +1,68 @@
+# Esoteric Language Virtual Machine
+
+## Registers and memory
+
+* Words are unsigned integers. The word-size is backend dependent, but
+  most backends use 24-bit words.
+* 6 registers: A, B, C, D, SP, and BP. They are one word wide and
+  initialized to zero.
+* Memory addresses are one word wide. The beginning of memory is
+  initialized from the .data segments; the rest is initialized to
+  zero.
+* Instructions are not stored in memory. Every instruction in a basic
+  block has the same pc (program counter) value, and a branch to a pc
+  goes to the first instruction with that pc.
+
+## Ops
+
+MOV dst, src
+- copy src to dst
+- src: immediate or register
+- dst: register
+
+ADD dst, src
+- add src to dst and places result in dst
+- src: immediate or register
+- dst: register
+- no carry flag
+
+SUB dst, src
+- subtract src from dst and places result in dst
+- src: immediate or register
+- dst: register
+- no borrow flag
+
+LOAD dst, src
+- copy address src to dst
+- src: immediate or register
+- dst: register
+
+STORE dst, src
```

Comment:
The direction is wrong here.

> STORE src, dst

I know this looks weird comparing to other operations, but I think this is a fairly common convention in RISC CPUs. This makes hardware implementation slightly easier, IIUC.